extension/js/common/message-renderer.ts

```diff
@@ -0,0 +1,795 @@
+    try {
+      if (['needChunk', 'maybePgp', 'publicKey'].includes(treatAs)) {
+        // todo: this isn't the best way to do this
+        // todo: move into a handler
+        // Inspect a chunk
+        if (this.debug) {
+          console.debug('processAttachment() try -> awaiting chunk + awaiting type');
+        }
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const data = await this.downloader.queueAttachmentChunkDownload(a).result;
+        const openpgpType = MsgUtil.type({ data });
+        if (openpgpType && openpgpType.type === 'publicKey' && openpgpType.armored) {
+          // if it looks like OpenPGP public key
+          treatAs = 'publicKey';
+        } else if (openpgpType && ['encryptedMsg', 'signedMsg'].includes(openpgpType.type)) {
+          treatAs = 'encryptedMsg';
+        } else {
+          if (this.debug) {
+            console.debug("processAttachment() try -> awaiting done and processed -- doesn't look like OpenPGP");
+          }
+          // plain attachment with a warning
+          loaderContext.renderPlainAttachment(a, attachmentSel, 'Unknown OpenPGP format');
+          return 'shown';
+        }
+        if (this.debug) {
+          console.debug('processAttachment() try -> awaiting done and processed');
+        }
+      }
+      if (treatAs === 'signature') {
+        // we could change 'Getting file info..' to 'Loading signed message..' in attachment_loader element
+        const raw = await this.msgGetRaw(msgId); // todo: can we try to restore the content attachment from 'full'?
+        loaderContext.hideAttachment(attachmentSel);
+        await this.setMsgBodyAndStartProcessing(loaderContext, 'signedDetached', messageInfo.printMailInfo, messageInfo.from?.email, renderModule =>
+          this.processMessageWithDetachedSignatureFromRaw(raw, renderModule, messageInfo.from?.email)
+        );
+        return 'hidden'; // native attachment should be hidden, the "attachment" goes to the message container
+      }
+      if (treatAs !== 'plainFile') {
+        loaderContext.hideAttachment(attachmentSel);
+      }
+      if (treatAs === 'hidden') {
+        return 'hidden';
+      } else if (treatAs === 'encryptedFile') {
+        // actual encrypted attachment - show it
+        loaderContext.prependEncryptedAttachment(a);
+        return 'replaced'; // native should be hidden, custom should appear instead
+      } else if (treatAs === 'encryptedMsg') {
+        await this.setMsgBodyAndStartProcessing(loaderContext, treatAs, messageInfo.printMailInfo, messageInfo.from?.email, (renderModule, frameId) =>
+          this.processCryptoMessage(a, renderModule, frameId, messageInfo.from?.email, messageInfo.isPwdMsgBasedOnMsgSnippet)
+        );
+        return 'hidden'; // native attachment should be hidden, the "attachment" goes to the message container
+      } else if (treatAs === 'publicKey') {
+        // todo - pubkey should be fetched in pgp_pubkey.js
```

Comment:
@tomholub This comment is about 5 years old.
We still pass public keys `embeddedPubkey` and private keys `embeddedBackup` in the query string.
Should we do something about it?

Reply:
I don't really see any issue, I'm unsure why the comment.

Reply:
I removed it